deepspeed/runtime/activation_checkpointing/checkpointing.py

```diff
@@ -311,6 +311,44 @@ def get_full_inputs(tensors, device=None):
     return tuple(inputs)
 
 
+def move_to_device(item, device):
+    if torch.is_tensor(item):
+        return item.to(device)
+    elif isinstance(item, list):
+        return [move_to_device(v, device) for v in item]
+    elif isinstance(item, tuple):
+        return tuple([move_to_device(v, device) for v in item])
+    elif isinstance(item, dict):
+        return {k: move_to_device(v, device) for k, v in item.items()}
+    else:
+        return item
+
+
+def filter_tensor_values(all_values):

```

Comment:
Super minor: the `value` in the names of the filter/merge routines first made me think we're filtering based on elements of the tensors (e.g., remove NaNs).  Maybe rename to something along the lines of `extract_tensors()` ?